diff --git a/vllm/v1/worker/gpu_model_runner.py b/vllm/v1/worker/gpu_model_runner.py
index 98e1dab..f192ec5 100644
--- a/vllm/v1/worker/gpu_model_runner.py
+++ b/vllm/v1/worker/gpu_model_runner.py
@@ -6372,11 +6372,6 @@ class GPUModelRunner(
             for layer_name in group.layer_names:
                 kv_cache = kv_caches[layer_name]
                 if isinstance(kv_cache_spec, AttentionSpec) and kv_cache.shape[0] == 2:
-                    assert kv_cache.shape[1] != 2, (
-                        "Fail to determine whether the layout is "
-                        "(2, num_blocks, ...) or (num_blocks, 2, ...) for "
-                        f"a tensor of shape {kv_cache.shape}"
-                    )
                     hidden_size = kv_cache.shape[2:].numel()
                     kv_cache.as_strided_(
                         size=kv_cache.shape,
